utils/storageverification/verify.go

```diff
@@ -0,0 +1,50 @@
+package storageverification
+
+import (
+	"cosmossdk.io/errors"
+	ibccommitmenttypes "github.com/cosmos/ibc-go/v8/modules/core/23-commitment/types"
+	"github.com/cosmos/ibc-go/v8/modules/core/exported"
+	ics23 "github.com/cosmos/ics23/go"
+
+	"github.com/neutron-org/neutron/v6/x/interchainqueries/types"
+)
+
+type VerifyCallback func(index int) error
+
+// VerifyStorageValues verifies stValues slice against proof using proofSpecs
+// A caller can provide verifyCallback method that will be called for each storage value from the slice with an index of the value in the slice
+// to do any additional user-defined checks of storage values
+func VerifyStorageValues(stValues []*types.StorageValue, root exported.Root, proofSpecs []*ics23.ProofSpec, verifyCallback VerifyCallback) error {
+	for index, value := range stValues {
+		proof, err := ibccommitmenttypes.ConvertProofs(value.Proof)
+		if err != nil {
+			return errors.Wrapf(ErrInvalidType, "failed to convert crypto.ProofOps to MerkleProof: %v", err)
+		}
+
+		if verifyCallback != nil {
+			if err := verifyCallback(index); err != nil {
+				return errors.Wrapf(ErrInvalidStorageValue, "failed to verify: %v", err)
```

Comment:
would be more clear what exactly failed
```suggestion
				return errors.Wrapf(ErrInvalidStorageValue, "verify callback failed at index %d: %v", index, err)
```